storage/innobase/dict/dict0dd.cc

```diff
@@ -635,6 +637,18 @@ dict_table_t *dd_table_create_on_dd_obj(const dd::Table *dd_table,
       case MYSQL_TYPE_TIMESTAMP2:
         field_type = MYSQL_TYPE_TIMESTAMP;
         break;
+      case MYSQL_TYPE_MEDIUM_BLOB:
+      case MYSQL_TYPE_LONG_BLOB:
+      case MYSQL_TYPE_TINY_BLOB:
+        field_type = MYSQL_TYPE_BLOB;
+        break;
+      case MYSQL_TYPE_NEWDATE:
+        field_type = MYSQL_TYPE_DATE;
+        break;
+      case MYSQL_TYPE_ENUM:
+      case MYSQL_TYPE_SET:
+        field_type = MYSQL_TYPE_STRING;
+        break;
```

Comment:
where is conversion done for regular running server??

Reply:
sql/field.h
 enum_field_types type() const final override { return MYSQL_TYPE_STRING; }